### PR TITLE
Add uml diagram for pin Command

### PR DESCRIPTION
**Pull request by prachi2023**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -81,6 +81,19 @@ the user input and execute the commands accordingly. The diagram below depicts t
 
 **Command**
 
+**PinCommand**
+
+1. Created by the parser function
```
Comment (R-Ramana) [inline docs/DeveloperGuide.md:86]: Don't think the PinCommand is created by the parser function right? Maybe can phrase it better?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -81,6 +81,19 @@ the user input and execute the commands accordingly. The diagram below depicts t
...
+1. Gets the note that is referenced too either by title or index 
+1. toggles the pinned status of the note
```
Comment (R-Ramana) [inline docs/DeveloperGuide.md:88]:
```suggestion
1. Toggles the pinned status of the specified note. 
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -81,6 +81,19 @@ the user input and execute the commands accordingly. The diagram below depicts t
 
 **Command**
 
+**PinCommand**
+
+1. Created by the parser function
+1. Gets the note that is referenced too either by title or index 
+1. toggles the pinned status of the note 
+1. returns the title as well as the pinned status of the note
```
Comment (R-Ramana) [inline docs/DeveloperGuide.md:89]:
```suggestion
1. Returns the title as well as the pinned status of the note. 
```

**Review by R-Ramana — APPROVED** [review]
Just some minor changes for your considerations on the Dev Guide.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (prachi2023) [reply docs/DeveloperGuide.md:86]: Instantiated by the parser function?

Comment (R-Ramana) [reply docs/DeveloperGuide.md:86]: Yeap I guess! But this one can leave out for now. Jun Xiang will be modifying the Parser command to make our code reflect more OOP

**Pull request merged**